pkg/vcdswaggerclient/api_access_levels.go

```diff
@@ -0,0 +1,266 @@
+
+/*
+AccessLevelsApiService Get the specified access level.
+Get the specified access level. 
+ * @param ctx context.Context - for authentication, logging, cancellation, deadlines, tracing, etc. Passed from http.Request or context.Background().
+ * @param accessLevelUrn
+
+@return AccessLevel
+*/
+func (a *AccessLevelsApiService) GetAccessLevel(ctx context.Context, accessLevelUrn string) (AccessLevel, *http.Response, error) {
+	var (
+		localVarHttpMethod = strings.ToUpper("Get")
+		localVarPostBody   interface{}
+		localVarFileName   string
+		localVarFileBytes  []byte
+		localVarReturnValue AccessLevel
+	)
+
+	// create path and map variables
+	localVarPath := a.client.cfg.BasePath + "/1.0.0/accessLevels/{accessLevelUrn}"
+	localVarPath = strings.Replace(localVarPath, "{"+"accessLevelUrn"+"}", fmt.Sprintf("%v", accessLevelUrn), -1)
+
+	localVarHeaderParams := make(map[string]string)
+	localVarQueryParams := url.Values{}
+	localVarFormParams := url.Values{}
+
+	// to determine the Content-Type header
+	localVarHttpContentTypes := []string{}
+
+	// set Content-Type header
+	localVarHttpContentType := selectHeaderContentType(localVarHttpContentTypes)
+	if localVarHttpContentType != "" {
+		localVarHeaderParams["Content-Type"] = localVarHttpContentType
+	}
+
+	// to determine the Accept header
+	localVarHttpHeaderAccepts := []string{"application/json;version=36.0"}
+
+	// set Accept header
+	localVarHttpHeaderAccept := selectHeaderAccept(localVarHttpHeaderAccepts)
+	if localVarHttpHeaderAccept != "" {
+		localVarHeaderParams["Accept"] = localVarHttpHeaderAccept
+	}
+	if ctx != nil {
+		// API Key Authentication
+		if auth, ok := ctx.Value(ContextAPIKey).(APIKey); ok {
+			var key string
+			if auth.Prefix != "" {
+				key = auth.Prefix + " " + auth.Key
+			} else {
+				key = auth.Key
+			}
+			localVarHeaderParams["Authorization"] = key
+			
+		}
+	}
+	r, err := a.client.prepareRequest(ctx, localVarPath, localVarHttpMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams, localVarFormParams, localVarFileName, localVarFileBytes)
+	if err != nil {
+		return localVarReturnValue, nil, err
+	}
+
+	localVarHttpResponse, err := a.client.callAPI(r)
+	if err != nil || localVarHttpResponse == nil {
+		return localVarReturnValue, localVarHttpResponse, err
+	}
+
+	localVarBody, err := ioutil.ReadAll(localVarHttpResponse.Body)
+	localVarHttpResponse.Body.Close()
+	if err != nil {
+		return localVarReturnValue, localVarHttpResponse, err
+	}
+
+	if localVarHttpResponse.StatusCode < 300 {
+		// If we succeed, return the data, otherwise pass on to decode error.
+		err = a.client.decode(&localVarReturnValue, localVarBody, localVarHttpResponse.Header.Get("Content-Type"));
+		return localVarReturnValue, localVarHttpResponse, err
+	}
+
+	if localVarHttpResponse.StatusCode >= 300 {
+		newErr := GenericSwaggerError{
+			body: localVarBody,
+			error: localVarHttpResponse.Status,
+		}
+		
+		if localVarHttpResponse.StatusCode == 200 {
```

Comment:
This pattern of checking if the code is >= 300 and then checking the status code to be 200 or 201 appears to be in multiple files. It appears that this "code == 200" of "code == 201" does not need to moved up since the <300 case already handles this logic. Therefore, this additional checking is just dead code. This would ideally be removed, but since there are several files, should we just leave this logic in the files?

Reply:
We can leave it for now, but I will create a ticket to handle this. There can be multiple potential solutions to this, including checking if swaggergen has a fix to this code and using that swaggergen to regenerate the 36.1 swagger client.

Created https://jira.eng.vmware.com/browse/VKCI-142 to work on this.